pkg/image/metrics/prometheus/metrics.go

```diff
@@ -0,0 +1,141 @@
+		metricSuccessCount,
+		"Counts successful image stream imports - both scheduled and not scheduled - per image registry",
+		[]string{labelScheduled, labelRegistry},
+		nil,
+	)
+	errorCountDesc = prometheus.NewDesc(
+		metricErrorCount,
+		"Counts number of failed image stream imports - both scheduled and not scheduled"+
+			" - per image registry and failure reason",
+		[]string{labelScheduled, labelRegistry, labelReason},
+		nil,
+	)
+
+	isc                 = importStatusCollector{}
+	registerLock        = sync.Mutex{}
+	collectorRegistered = false
+)
+
+type importStatusCollector struct {
+	cbCollectISCounts        QueuedImageStreamFetcher
+	cbCollectScheduledCounts QueuedImageStreamFetcher
+}
+
+// InitializeImportCollector is supposed to be called by image import controllers when they are prepared to
+// serve requests. Once all the controllers register their callbacks, the collector registers the metrics with
+// the prometheus.
+func InitializeImportCollector(
+	scheduled bool,
+	cbCollectISCounts QueuedImageStreamFetcher,
+) {
+	registerLock.Lock()
+	defer registerLock.Unlock()
+
+	if scheduled {
+		isc.cbCollectScheduledCounts = cbCollectISCounts
+	} else {
+		isc.cbCollectISCounts = cbCollectISCounts
+	}
+
+	if collectorRegistered {
+		return
+	}
+
+	if isc.cbCollectISCounts != nil && isc.cbCollectScheduledCounts != nil {
+		prometheus.MustRegister(&isc)
+		collectorRegistered = true
+		glog.V(4).Info("Image import controller metrics registered with prometherus")
+	}
+}
+
+func (isc *importStatusCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- successCountDesc
+	ch <- errorCountDesc
+}
+
+func (isc *importStatusCollector) Collect(ch chan<- prometheus.Metric) {
+	successCounts, errorCounts, err := isc.cbCollectISCounts()
+	if err != nil {
+		glog.Errorf("Failed to collect scheduled image import metrics: %v", err)
+		ch <- prometheus.NewInvalidMetric(successCountDesc, err)
+		// allow for partial scrub ??
+		// return
+	} else {
+		pushSuccessCounts("false", successCounts, ch)
+		pushErrorCounts("false", errorCounts, ch)
+	}
+
+	successCounts, errorCounts, err = isc.cbCollectScheduledCounts()
+	if err != nil {
+		glog.Errorf("Failed to collect scheduled image import metrics: %v", err)
+		ch <- prometheus.NewInvalidMetric(errorCountDesc, err)
+		return
+	}
+
+	pushSuccessCounts("true", successCounts, ch)
+	pushErrorCounts("true", errorCounts, ch)
+}
+
+func pushSuccessCounts(scheduled string, counts ImportSuccessCounts, ch chan<- prometheus.Metric) {
+	for registry, count := range counts {
+		ch <- prometheus.MustNewConstMetric(
+			successCountDesc,
+			prometheus.CounterValue,
+			float64(count),
+			scheduled,
+			registry)
+	}
+}
+
+func pushErrorCounts(scheduled string, counts ImportErrorCounts, ch chan<- prometheus.Metric) {
+	for info, count := range counts {
+		ch <- prometheus.MustNewConstMetric(
+			errorCountDesc,
+			prometheus.CounterValue,
+			float64(count),
+			scheduled,
+			info.Registry,
+			info.Reason)
```

Comment:
@gabemontero isn't this the failure reason label you were asking about?

Reply:
totally missed that ... yep ignore my comment